presto-hive/src/main/java/com/facebook/presto/hive/IonSqlQueryBuilder.java

```diff
@@ -0,0 +1,310 @@
+                        rangeConjuncts.add(toPredicate(">", range.getLow().getValue(), type, position));
+                        break;
+                    case EXACTLY:
+                        rangeConjuncts.add(toPredicate(">=", range.getLow().getValue(), type, position));
+                        break;
+                    case BELOW:
+                        throw new IllegalArgumentException("Low marker should never use BELOW bound");
+                    default:
+                        throw new AssertionError("Unhandled bound: " + range.getLow().getBound());
+                }
+            }
+            if (!range.getHigh().isUpperUnbounded()) {
+                switch (range.getHigh().getBound()) {
+                    case ABOVE:
+                        throw new IllegalArgumentException("High marker should never use ABOVE bound");
+                    case EXACTLY:
+                        rangeConjuncts.add(toPredicate("<=", range.getHigh().getValue(), type, position));
+                        break;
+                    case BELOW:
+                        rangeConjuncts.add(toPredicate("<", range.getHigh().getValue(), type, position));
+                        break;
+                    default:
+                        throw new AssertionError("Unhandled bound: " + range.getHigh().getBound());
+                }
+            }
+            // If rangeConjuncts is null, then the range was ALL, which should already have been checked for
+            checkState(!rangeConjuncts.isEmpty());
+            disjuncts.add("(" + Joiner.on(" AND ").join(rangeConjuncts) + ")");
+        }
+
+        // Add back all of the possible single values either as an equality or an IN predicate
+        if (singleValues.size() == 1) {
+            disjuncts.add(toPredicate("=", getOnlyElement(singleValues), type, position));
+        }
+        else if (singleValues.size() > 1) {
+            List<String> values = new ArrayList<>();
+            for (Object value : singleValues) {
+                checkType(type);
+                values.add(valueToQueryString(type, value));
+            }
+            disjuncts.add(createColumnString(type, position) + " IN (" + Joiner.on(",").join(values) + ")");
+        }
+
+        // Add nullability disjuncts
+        checkState(!disjuncts.isEmpty());
+        if (domain.isNullAllowed()) {
+            disjuncts.add(format("s._%d", position + 1) + " = '' ");
+        }
+
+        return "(" + Joiner.on(" OR ").join(disjuncts) + ")";
+    }
+
+    private String toPredicate(String operator, Object value, Type type, int position)
+    {
+        checkType(type);
+
+        return format("%s %s %s", createColumnString(type, position), operator, valueToQueryString(type, value));
+    }
+
+    private static void checkType(Type type)
+    {
+        checkArgument(isSupportedType(type), "Type not supported: %s", type);
+    }
+
+    private static String valueToQueryString(Type type, Object value)
```

Comment:
What do you think about renaming this to `value()` and the `createColumnString()` method to `column()`? The names of these methods contain the word "string", which is obvious from the return value. I feel like with these names we don't lose readability, but make it simpler and somewhat more fluent when reading the code. If you think this is less readable then please at least make the naming consistent: `createValueString` and `createColumnString` for example.

Reply:
I agree, this is one of the principles given in [Clean Code](https://www.oreilly.com/library/view/clean-code/9780136083238/). Refactored.